lib/ansiblelint/utils.py

```diff
@@ -573,3 +575,82 @@ def construct_mapping(node, deep=False):
     except (yaml.parser.ParserError, yaml.scanner.ScannerError) as e:
         raise SystemExit("Failed to parse YAML in %s: %s" % (filename, str(e)))
     return data
+
+
+def add_skipped_rules(orig_file_text, file_type):
+    yaml = ruamel.yaml.YAML()
+    data = yaml.load(orig_file_text)
+
+    if not data:
+        return orig_file_text
+
+    if file_type in ('tasks', 'handlers'):
+        tasks = data
+    elif file_type == 'playbook':
+        # this does look at values in top-level dict in playbook.yml
+        # example "sudo: True" outside "tasks" not looked at for skipping
+        if not (isinstance(data, list) and isinstance(data[0], dict)):
+            return orig_file_text
+        tasks = data[0].get('tasks', None)
+    elif file_type == 'meta':
+        if isinstance(data, dict):
+            tasks = [data]
+        else:
+            tasks = data
+    else:
+        return orig_file_text
+
+    if not tasks:
+        return orig_file_text
+
+    any_skipped_rules = False
+    for task in tasks:
+        skipped_rules = _get_rule_skips_from_task(task)
+        if skipped_rules:
+            task['skipped_rules'] = skipped_rules
+            any_skipped_rules = True
+
+    if not any_skipped_rules:
+        return orig_file_text
+
+    # dump to file with new skipped_rules list(s), read file and return text
+    with tempfile.NamedTemporaryFile('w') as tf:
```

Comment:
Use in-memory file-like buffers instead.

Reply:
You can rely on `six.StringIO` (or `six.BytesIO`?)

It should look like:
```python
import six
with six.StringIO() as in_memory_file:
    yaml.dump(data, in_memory_file)
    in_memory_file.seek(0)
    new_file_text = in_memory_file.read()
```

(needs testing)

Reply:
`io.BytesIO()` worked in py2 and py3